hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSDirectory.java

```diff
@@ -160,6 +164,8 @@ private static INodeDirectory createRoot(FSNamesystem namesystem) {
   private final int contentCountLimit; // max content summary counts per run
   private final long contentSleepMicroSec;
   private final INodeMap inodeMap; // Synchronized by dirLock
+  // Temp InodeMap used when loading an FS image.
+  private final GSet<INode, INodeWithAdditionalFields> inodeMapTemp;
```

Comment:
Can this be implemented so that `inodeMapTemp` is local variable, rather than a member of the class? It is used only during loading, and we don't need to keep it permanently after loading is done.

Reply:
There are three classes that work together to load an FS image.
_FSImage.java_ uses functions from _FSImageFormatProtobuf.java_, which uses functions from  _FSImageFormatPBINode.java_. If we make inodeMapTemp a local variable in loadFSImage@676 in _FSImage.java_, potentially, we need to pass that variable down the stream and need to modify quite extensively. I am not sure we want to do that.

Right now, _FSImageFormatPBINode_ has a reference to the FSDirectory object and thus can call functions of FSDirectory, to add inodes directly.

Reply:
agreed to stay as it is, as the proposed change would require changes to dozens of functions across three files.